Signup

HashamReyaz972 opened the pull request:
Created  API for signing users in.

Current Features:
1.  Checks if user exists in masterSchema and adds him in a new Login collection.
2.  Makes sure same email isn't used for multiple users.

prakritidev committed:
setup: added libs, Eslint initialization, updated readme.md.

prakritidev committed:
lib-installation: nodemon

prakritidev committed:
containerization: added docker and  dockercompose file.

prakritidev committed:
update: node index.js to nodemon index.js

prakritidev committed:
update: updated .gitignore

prakritidev committed:
added configs for logger, env, docker, mongo, redis, aws.

prakritidev committed:
feat: tested mongodb and express server

Commit:
feat:  removed eslint

Commit:
Created a Signup API

prakritidev commented on routes/routes.js:
```diff
@@ -0,0 +1,19 @@
+   db
+}) => {
+   router
+      .route('/employee/signup')
```
.route('/signup')

prakritidev commented on routes/routes.js:
```diff
@@ -0,0 +1,19 @@
+   router
+      .route('/employee/signup')
+      .post((req, res, next) => {
+         return employeeController.signupEmployee(req, res, next, {
```
function name shoutlf be same as APIs name.

prakritidev commented on services/services.signupEmployee.js:
```diff
@@ -0,0 +1,54 @@
+
+module.exports = () => {
+   /**
+    * 
```
function description.

prakritidev commented on services/services.signupEmployee.js:
```diff
@@ -0,0 +1,54 @@
+
+         //Check to see if the employee with same user_id is present in masterdata
+         let matchedEmployee = Masterdata.findOne({
+            user_id
```
use const instead of let. use response_usecase for receiving a response from any service.

prakritidev commented on services/services.signupEmployee.js:
```diff
@@ -0,0 +1,54 @@
+const Login = require('../model/loginSchema');
```
const user = require('../model/loginSchema');

prakritidev commented on services/services.signupEmployee.js:
```diff
@@ -0,0 +1,54 @@
+         });
+         if (matchedEmployee) {
+            //If the employee id matches, then add him to the login collection(sign him up)
+            matchedEmployee = new Login({
```
const user = new User({})

prakritidev commented on services/services.signupEmployee.js:
```diff
@@ -0,0 +1,54 @@
+               company,
+            })
+            //Check if email used is already taken
+            if (await Login.findOne({
```
move it up.

prakritidev commented on model/masterSchema.js:
```diff
@@ -0,0 +1,9 @@
+const mongoose = require('mongoose');
+
+const {
```
check ESLINT  config.

prakritidev commented on model/loginSchema.js:
```diff
@@ -0,0 +1,29 @@
+const loginSchema = new Schema({
+   user_id: {
+      type: String,
+      unique: true,
```
Add messege.

prakritidev commented on controller/controller.Employee.js:
```diff
@@ -0,0 +1,49 @@
+           const response = await signupEmployeeService.signupEmployee(payload, logger, db);
+           if (response == 'foundEmail') {
+              res.status(200).send({
+                 status: '200',
```
status: '200 OK',

prakritidev commented on controller/controller.Employee.js:
```diff
@@ -0,0 +1,49 @@
+           if (response == 'foundEmail') {
+              res.status(200).send({
+                 status: '200',
+                 result: 'Email Id Already Exists'
```
better response message. 

EX: ${payload.email} already exists in the database.

prakritidev commented on controller/controller.Employee.js:
```diff
@@ -0,0 +1,49 @@
+           } else {
+              res.status(200).send({
+                 status: 400,
+                 msg: 'Invalid Data'
```
Give the exact reason.what is invalid.

prakritidev commented on controller/controller.Employee.js:
```diff
@@ -0,0 +1,49 @@
+                 msg: 'Invalid Data'
+              });
+           }
+           logger.info('Signin successful');
```
beeter response.

prakritidev replied:
sedn user name as well.

Commit:
Code Cleanup:
1.FIle names and fuction names changed.
2.Success and error messages improved
3.Made sure 'user_id' can be recieved by the api as a string too.

Co-authored-by: Prakritidev Verma <Prakritidevverma@gmail.com>
